Java.md

```diff
@@ -1 +1,201 @@
-# Java
\ No newline at end of file
+# Java
+
+## Class
+
+### Layout
+
+Methods and fields should be organized as below:
+
+```java
+class MySuperbClass {
+	// Fields
+	
+	// Constructors
+	
+	// Private & protected methods
+	
+	// Public methods
+}
+```
+
+### Indentation
+
+Standard indentation should be **4 soft spaces**. When a carriage return is needed, keep same indentation.
+
+```java
+class AnotherAmazingClass {
+    public void aMethod() {
+        anotherMethod(
+            firstArg,
+            secondArg,
+            thirdArg,
+            // ...
+        );
+        
+        stillAnotherMethod();
+    }
+}
+```
+
+### Names
+
+Class and method names should be the most limpid as possible. Do not worry about having a long class name. Auto-completion and refactor tools are there for helping you.
+
+Prefer `sortUsersByNameAsc` to `getUsers`.
+
+### Case
+
+In Java world, methods and vars should use **camel case**: `myFantasticMethod` . Class should be **pascal case**: `FriendlyClass`.
+
+Besides, it is a good practice to prefix private and protected methods and fields **by an underscore**. In no time at all, you can easily spot the access of your field/method.
+
+```java
+public class User {
+    private String _name;
+    
+    private String _engineString(String value) {
+        // ...
+    }
+    
+    public String getName() {
+        // ...
+    }
+}
+```
+
+## Interface
+
+### Name
+
+Interface name should start with a capital `I` to make distinction easier with classes.
+
+## Performances
+
+### For loops
+
+Use **foreach** loop as much as possible, for it is optimized. Otherwise, when using a classic **for** loop, save size of browsable element before running:
+
+```java
+// DO
+
+int size = list.size();
+for (int i = 0; i < size; i++) {
+    // ...
+}
+
+// DO NOT
+
+for (int i = 0; i < list.size(); i++) {
+    // ...
+}
+```
+
+Do not assume size access is constant.
+
+### Variable declarations
+
+Variables should be declared at the top of each block. This process increases performances for it avoids calling memory allocator and garbage collector again.
```

Comment:
I thought we discussed this and determined that it doesn't really affect performance in modern JVMs.